### PR TITLE
PR: Implement support for modern naming convention and aliases specification from the config mapping file.

### LINKED ISSUE #44: Add aliases to the configs
In OCIO v2, a colorspace may have an aliases attribute that provides alternate names for the color space.  This allows configs to be more compact than the earlier ACES configs (which duplicated each color space to define the alternate name).

The spreadsheet of color spaces has a column for the alias strings (although they may need to be refined).  In the Python, these strings may be provided as a list in the ColorSpace constructor or later individually using the addAlias method.

It seems there is code to add these at line 194 in: 
[opencolorio_config_aces/config/generation/factories.py](https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/blob/d588eac7fc453ea55454fc7f04ad71bac85b339e/opencolorio_config_aces/config/generation/factories.py)

But the aliases don't seem to be showing up in the config.  Is it an issue in the code or the spreadsheet or something else?

### PR DESCRIPTION
**Pull request by KelSolaar**
This PR is a first pass at addressing #57 and also closes #44.

The main difference with @doug-walker's original proposal is that instead of having a colorspace named `Linear sRGB`, it is named `sRGB - Linear` (I could not stand it being like that when we have `sRGB - Display`, `sRGB - Texture`, etc...)

Some names require a generator update as they are directly inherited from the *Reference* config generation, some others, e.g. CLF are pulled down from the config mapping file, I reckon it might be better to have that done programmatically too.

There is still quite a bit of naming to be done, and some family business to do.

I did clean the code quite a bit and added a lot more logging doing so.

### Reference Config Colorspace Names

- Blue Light Artifact Fix
- ACES 1.3 Reference Gamut Compression
- ACES 1.0 - SDR Video
- ACES 1.0 - SDR Video (D60 sim on D65)
- ACES 1.1 - SDR Video (P3 lim)
- ACES 1.1 - SDR Video (Rec.709 lim)
- ACES 1.1 - HDR Video (1000 nits & Rec.2020 lim)
- ACES 1.1 - HDR Video (2000 nits & Rec.2020 lim)
- ACES 1.1 - HDR Video (4000 nits & Rec.2020 lim)
- ACES 1.1 - HDR Video (1000 nits & P3 lim)
- ACES 1.1 - HDR Video (2000 nits & P3 lim)
- ACES 1.1 - HDR Video (4000 nits & P3 lim)
- ACES 1.0 - SDR Cinema
- ACES 1.1 - SDR Cinema (D60 sim on D65)
- ACES 1.1 - SDR Cinema (Rec.709 lim)
- ACES 1.0 - SDR Cinema (D60 sim on DCI)
- ACES 1.1 - SDR Cinema (D65 sim on DCI)
- ACES 1.1 - HDR Cinema (108 nits & P3 lim)
- Un-tone-mapped
- CIE-XYZ-D65
- sRGB - Display
- Rec.1886 / Rec.709 Video - Display
- Rec.1886 / Rec.2020 Video - Display
- Rec.2100-HLG - Display
- Rec.2100-PQ - Display
- ST2084-P3-D65 - Display
- P3-D60 - Display
- P3-D65 - Display
- P3-DCI - Display
- ACES2065-1
- Raw
- ACEScc
- ACEScct
- ACEScg
- ADX10
- ADX16
- CLog2 CGamut
- CLog3 CGamut
- Log3G10 RWG
- LogC EI800 AWG
- SLog3 SGamut3
- SLog3 SGamut3Cine
- SLog3 Venice SGamut3
- SLog3 Venice SGamut3Cine
- VLog VGamut

### CG Config Names

- ACES 1.0 - SDR Video
- ACES 1.1 - HDR Video (1000 nits & Rec.2020 lim)
- ACES 1.1 - HDR Video (1000 nits & P3 lim)
- ACES 1.0 - SDR Cinema
- Un-tone-mapped
- CIE-XYZ-D65
- sRGB - Display
- Rec.1886 / Rec.709 Video - Display
- Rec.2100-PQ - Display
- ST2084-P3-D65 - Display
- P3-D65 - Display
- ACES2065-1
- Raw
- ACEScc
- ACEScct
- ACEScg
- sRGB - Texture
- P3-D65 - Linear
- Rec.2020 - Linear
- Rec.709 - Linear
- sRGB - Linear
- Gamma 1.8 - Rec.709 - Texture
- Gamma 2.2 - AP1 - Texture
- Gamma 2.2 - Rec.709 - Texture
- Gamma 2.4 - Rec.709 - Texture
- Rec.1886 - Curve
- sRGB - Curve

### CODE REVIEW

**Comment by KelSolaar** [comment]
The build artefacts are here btw: https://github.com/AcademySoftwareFoundation/OpenColorIO-Config-ACES/actions/runs/2639340083

**Review by doug-walker — COMMENTED** [review]
Thanks for this @KelSolaar !  

There is a minor issue in your CG list above you have "Linear to Rec.1886" and "Linear to sRGB" in addition to "Rec.1886 - Curve" and "sRGB - Curve". 

Regarding the "Linear" issue, let's discuss more at the meeting.  One point I will make now is that for color spaces that do not have an official name, we have been following a pattern of Transfer Function followed by Primaries.  So I put Linear first to follow that pattern.  Furthermore, "Texture", "Display", and "Curve" are really more adjectives (unessential but helpful, and therefore may go at the end) whereas "Linear" is more like a noun (essential info).

**Comment by doug-walker** [comment]
Updates are needed to the aliases list in the spreadsheet to capture the 1.2 config names ("ACES - ACES2065-1", etc.).  One question for the group: I'm noticing there are aliases defined for roles such as "Role - scene_linear".  Do we really want to define those as aliases?

**Comment by KelSolaar** [comment]
We are using the `Role - .*` colorspaces.

**Comment by doug-walker** [comment]
To discuss, but I think we should avoid introducing color space or alias names that are ambiguous and likely to change over time or be different between facilities/shows.  There has been a lot of interest in the idea of being able to merge configs or convert color spaces between configs.  It's straight forward to handle the "many names for one color space" problem, but the "several different color spaces with the same name" problem is much harder to handle.

File: opencolorio_config_aces/config/cg/generate/config.py [hunk 1]
```diff
@@ -353,11 +357,35 @@ def generate_config_cg(
...
     """
     Generate the *ACES* Computer Graphics (CG) *OpenColorIO* config.
 
+    The default process is as follows:
+
+    -   The *ACES* CG *OpenColorIO* config generator invokes the *aces-dev*
+        reference implementation *OpenColorIO* config generator via the
+        :func:`opencolorio_config_aces.generate_config_aces` definition and the
+        default reference config mapping file.
+    -   The *ACES* CG *OpenColorIO* config generator filters and and extend
```
Comment (doug-walker) [inline opencolorio_config_aces/config/cg/generate/config.py:372]: "and and extend" --> "and extends"

File: opencolorio_config_aces/config/cg/generate/config.py [hunk 2]
```diff
@@ -262,7 +266,7 @@ def config_basename_cg(
 ):
     """
     Generate the ACES* Computer Graphics (CG) *OpenColorIO* config
-    basename, i.e. the filename devoid of directory prefix.
+    basename, i.e. the filename devoid of directory affixe.
```
Comment (doug-walker) [inline opencolorio_config_aces/config/cg/generate/config.py:269]: affixe --> affix

**Review by doug-walker — DISMISSED** [review]
I think we agreed at the last working group meeting that we would go ahead and merge this to help streamline the development process, even though we are still working through some details of the naming.

File: opencolorio_config_aces/config/cg/generate/config.py (hunk 1 above)
Comment (KelSolaar) [reply opencolorio_config_aces/config/cg/generate/config.py:372]: Thanks! Little known but powerful feature:

image

```suggestion
    -   The *ACES* CG *OpenColorIO* config generator filters and and extends
```

File: opencolorio_config_aces/config/cg/generate/config.py (hunk 2 above)
Comment (KelSolaar) [reply opencolorio_config_aces/config/cg/generate/config.py:269]:
```suggestion
    basename, i.e. the filename devoid of directory affix.
```

**Comment by KelSolaar** [comment]
Thanks @doug-walker! I did the small changes.

**Review by doug-walker — APPROVED** [review]

**Pull request merged**